kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/channel/Channel.java

```diff
@@ -142,20 +210,60 @@ public boolean isEnabled() {
      *
      * @return a double that represents the scale factor to be applied to the read value
      *
+     * @throws IllegalArgumentException
+     * 
      * @since 2.3
+     * 
+     * @deprecated Use {@link #getValueScaleAsTypedValue()}
      */
+    @Deprecated
     public double getValueScale() {
+        if (this.scaleOffsetType != ScaleOffsetType.DEFINED_BY_VALUE_TYPE) {
+            throw new IllegalStateException("the type of the scale is " + this.getScaleOffsetType().toString());
+        }
+        return (Double) this.valueScale;
+    }
+
+    /**
+     * Returns a {@link Number} that represents the scale factor to be applied to the read
+     * value
+     *
+     * @return a {@link Number} that represents the scale factor to be applied to the read value
+     *
+     * @since 2.8
+     */
+    public Number getValueScaleAsNumber() {
         return this.valueScale;
     }
 
     /**
      * Returns a double that represents the offset to be applied to the read value
      *
      * @return a double that represents the offset to be applied to the read value
+     * 
+     * @throws IllegalArgumentException
      *
      * @since 2.3
+     * 
+     * @deprecated Use {@link #getValueOffsetAsTypedValue()}
      */
+    @Deprecated
     public double getValueOffset() {
+        if (this.scaleOffsetType != ScaleOffsetType.DEFINED_BY_VALUE_TYPE) {
```

Comment:
Same as above for the scale

Reply:
it's an illegal state of the object, i would throw the exception because using the method results in an unintentional casting

Reply:
i'm reconsidering this, an user needs to change the api in a client application to not break it loading a configuraiton without the scaleoffset value property, i'll remove the exception throwing